dir/fs.go

```diff
@@ -92,10 +90,16 @@ func (u unionDirFS) GetPath(elem ...string) (string, error) {
 		}
 		// got the first existing file path and break
 		// return the path with current OS separator
-		return targetPath, nil
+		return filepath.Join(dir.Root(), pathSuffix), nil
 	}
-	// return the last possible path for creating new file
-	return targetPath, &fs.PathError{
+	// if the given path does not exist, return the first possible path in the
+	// union directories for creating new file.
+	var fallbackPath string
+	if len(u.Dirs) > 0 {
```

Comment:
Can we throw an error if `u.Dirs` is empty?

Reply:
Yes, added an error for empty `u.Dirs`.